### PR TITLE
Clients connected to container event stream now receive the same data

### PR DESCRIPTION
**Pull request by Freggy**
#### What type of PR is this?

/kind feature

#### What this PR does / why we need it:
Enables multiple connected clients to the `ContainerEvent` stream to receive the same data.

#### Which issue(s) this PR fixes:
Fixes https://github.com/cri-o/cri-o/issues/6466

#### Does this PR introduce a user-facing change?

```release-note
None
```

### CODE REVIEW

**Comment by haircommander** [comment]
Thanks! From a quick skim on mobile the change LGTM. I will do a more thorough review tomorrow. In the meantime
/ok-to-test
/cc @sairameshv @harche

**Comment by Freggy** [comment]
Alright, I will fix the linting issue tomorrow. Is there anything else to look out for regarding tests etc?

**Comment by haircommander** [comment]
the unit tests you have now are sufficient. when (eventually) `crictl events` is added, we can address adding integration tests (which live in the test/ directory)

**Comment by Freggy** [comment]
linting issues should now be fixed. if you need me to do anything else just let me know :)

**Comment by haircommander** [comment]
/approve

LGTM, @harche or @sairameshv can you review for the final `/lgtm`

File: server/container_events.go [hunk 1]
```diff
@@ -1,17 +1,76 @@
...
-// GetContainerEvents sends the stream of container events to the client - kubelet
-func (s *Server) GetContainerEvents(req *types.GetEventsRequest, ces types.RuntimeService_GetContainerEventsServer) error {
-	if s.Config().EnablePodEvents {
-		for containerEvent := range s.ContainerEventsChan {
-			if err := ces.Send(&containerEvent); err != nil {
-				return err
-			}
+type containerEventConn struct {
+	Wg  sync.WaitGroup
+	Err error
```
Comment (sairameshv) [inline server/container_events.go:13]: Thanks for the PR @Freggy !!
Changes LGTM expect that do we need to export these fields if not used in any other package?

Comment (Freggy) [reply server/container_events.go]: No problem, I'm hoping to contribute more in the future :) 
There is no need for those fields being public, so I made them private.

**Comment by sairameshv** [comment]
@Freggy , Could you squash all the commits into a single commit?

**Comment by Freggy** [comment]
Done @sairameshv

**Pull request merged**